backs/randomizer.go

```diff
@@ -2,71 +2,72 @@ package backs
 
 import (
 	"fmt"
+	"io/fs"
 	"math/rand"
-	"os"
+	"path"
+	"strings"
 )
 
-type backRarity struct {
-	Type  string
-	Value int
+type back struct {
+	path string
 }
 
-func getRarities() []backRarity {
-	return []backRarity{
-		{Type: "Rollback", Value: 1},
-		{Type: "Rare", Value: 10},
-		{Type: "Uncommon", Value: 90},
-		{Type: "Common", Value: 400},
-	}
+func (b back) Filename() string {
+	return path.Base(b.path)
 }
 
-func maxRarities() int {
-	max := 0
-	for _, rarity := range getRarities() {
-		if rarity.Value > max {
-			max = rarity.Value
-		}
-	}
-	return max
+func (b back) Backname() string {
+	return strings.Split(b.Filename(), ".")[0]
 }
 
-type BackMapping map[string][]string
+func (b back) Rarity() rarity {
+	r, _ := lookUpRarity(path.Base(path.Dir(b.path)))
+	return r
+}
+
+type BackMapping map[rarity][]back
 
 // GetBacks gets all the file paths assigned to their rarities
-func GetBacks() (*BackMapping, error) {
```

Comment:
Forgot to mention this in the PR description, but I removed pointers to the map type because maps are a "reference type" that consist of internal pointers (i.e. two copies of the same map variable share underlying memory, even though those variables themselves are not pointer types)

Reply:
Hell it happens. Sorry for mansplaining reference types